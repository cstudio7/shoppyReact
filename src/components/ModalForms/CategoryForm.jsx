import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button, Form } from 'semantic-ui-react';
import { updateCategory } from '../../actions/updateCategoryAction';
import { addCategory } from '../../actions/addCategoryAction';
import { productCategories } from '../../actions/getCategoriesAction';
import './ModalForms.scss';

class CategoryForm extends Component {
  state = {
    categoryDetails: {
      name: this.props.details.nameToDisplay || '',
      image_url: this.props.details.imageUrl || '',
      description: this.props.details.description || ''
    }
  };

  componentWillUnmount = () => {
    clearTimeout(this.state.timeoutId);
  };

  handleFormChange = (event) => {
    const { categoryDetails } = this.state;
    const { name, value } = event.target;
    categoryDetails[name] = value;
    this.setState({
      categoryDetails
    });
  };

  handleFormSubmit = (event) => {
    event.preventDefault();
    const { categoryDetails } = this.state;
    const { id } = this.props.details;
    if (Object.keys(this.props.details).length === 0) {
      this.props.addCategory(categoryDetails);
    } else {
      this.props.updateCategory(categoryDetails, id);
    }
    const timeoutId = setTimeout(() => this.props.productCategories(), 500);
    this.setState({ timeoutId });
  };

  render() {
    const { name, image_url, description } = this.state.categoryDetails;
    return (
      <Form className="modalForm" onSubmit={this.handleFormSubmit}>
        <Form.Field className="formField">
          <input
            placeholder="Category Name"
            value={name}
            onChange={this.handleFormChange}
            name="name"
          />
        </Form.Field>
        <Form.Field className="formField">
          <input
            placeholder="Image URL (example:https://exampleimageurl.jpg)"
            typeof="url"
            value={image_url}
            onChange={this.handleFormChange}
            name="image_url"
          />
          <a target="_blank" rel="noopener noreferrer" href="https://imgur.com/">
            Upload image here
          </a>
        </Form.Field>
        <Form.TextArea
          placeholder="Short Description"
          className="formField"
          value={description}
          onChange={this.handleFormChange}
          name="description"
        />
        <Button type="submit">Submit</Button>
      </Form>
    );
  }
}

const mapStateToProps = state => ({
  state
});

export { CategoryForm as CategoryFormPage };
export default connect(
  mapStateToProps,
  { updateCategory, addCategory, productCategories }
)(CategoryForm);
