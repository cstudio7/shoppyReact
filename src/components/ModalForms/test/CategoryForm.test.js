import React from 'react';
import { shallow } from 'enzyme';
import CategoryForm from '../CategoryForm';

describe('Test the CategoryForm Component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<CategoryForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
