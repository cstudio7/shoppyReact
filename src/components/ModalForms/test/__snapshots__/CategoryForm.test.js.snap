// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the CategoryForm Component should match snapshot 1`] = `
<Form
  as="form"
  className="modalForm"
  onSubmit={[Function]}
>
  <FormField
    className="formField"
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Category Name"
      value="pname"
    />
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="image_url"
      onChange={[Function]}
      placeholder="Image URL (example:https://exampleimageurl.jpg)"
      typeof="url"
      value="https://i.imgur.com/27DUH5b.jpg"
    />
    <a
      href="https://imgur.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Upload image here
    </a>
  </FormField>
  <FormTextArea
    as={[Function]}
    className="formField"
    control={[Function]}
    name="description"
    onChange={[Function]}
    placeholder="Short Description"
    value="admin@mail.com"
  />
  <Button
    as="button"
    type="submit"
  >
    Submit
  </Button>
</Form>
`;

exports[`Test the CategoryForm Component should match snapshot 2`] = `
<Form
  as="form"
  className="modalForm"
  onSubmit={[Function]}
>
  <FormField
    className="formField"
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Category Name"
      value="pname"
    />
  </FormField>
  <FormField
    className="formField"
  >
    <input
      name="image_url"
      onChange={[Function]}
      placeholder="Image URL (example:https://exampleimageurl.jpg)"
      typeof="url"
      value=""
    />
    <a
      href="https://imgur.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Upload image here
    </a>
  </FormField>
  <FormTextArea
    as={[Function]}
    className="formField"
    control={[Function]}
    name="description"
    onChange={[Function]}
    placeholder="Short Description"
    value=""
  />
  <Button
    as="button"
    type="submit"
  >
    Submit
  </Button>
</Form>
`;
